packages/python-packages/api-stub-generator/apistub/_stub_generator.py

```diff
@@ -0,0 +1,210 @@
+            if module_nav:
+                navigation.add_child(module_nav)
+        return apiview
+
+
+    def _serialize_tokens(self, apiview, out_file_path):
+        """Serialize tokens into json file
+        :param ApiView: apiview
+        """
+        # Serialize apiview to json
+        logging.info("Serializing tokens into json")
+        json_apiview = APIViewEncoder().encode(apiview)
+        with open(out_file_path, "w") as json_file:
+            json_file.write(json_apiview)
+        logging.info("Serialized tokens into json file [{}]".format(out_file_path))
+
+
+    def _extract_wheel(self):
+        """Extract the wheel into out dir and return root path to azure root directory in package
+        """
+        file_name, _ = os.path.splitext(os.path.basename(self.pkg_path))
+        temp_pkg_dir = os.path.join(self.temp_path, file_name)
+        if os.path.exists(temp_pkg_dir):
+            logging.info("Cleaning up existing temp directory: {}".format(temp_pkg_dir))
+            shutil.rmtree(temp_pkg_dir)
+        os.mkdir(temp_pkg_dir)
+
+        logging.info("Extracting {0} to directory {1}".format(self.pkg_path, temp_pkg_dir))
+        zip_file = zipfile.ZipFile(self.pkg_path)
+        zip_file.extractall(temp_pkg_dir)
+        logging.info("Extracted package files into temp path")
+        return temp_pkg_dir
+
+
+    def _parse_pkg_name(self):
+        file_name = os.path.basename(self.pkg_path)
+        whl_name, extn = os.path.splitext(file_name)
+        if extn[1:] not in ['whl', 'zip']:
+            raise ValueError("Invalid type of package. API view parser expects wheel or sdist package")
+
+        filename_parts = whl_name.split("-")
+        pkg_name = filename_parts[0].replace("_", "-")
+        version = filename_parts[1]
+        return pkg_name, version
+
+
+    def _install_package(self, pkg_name):
+        # Uninstall the package and reinstall it to parse so inspect can get members in package
+        # We don't want to force reinstall to avoid reinstalling other dependent packages
+        commands = [sys.executable, "-m", "pip", "uninstall", pkg_name, "--yes"]
+        check_call(commands)
+        commands = [sys.executable, "-m", "pip", "install", self.pkg_path]
+        check_call(commands)
+
+
+    def generate_tokens(self):
+        # Extract package to temp directory
+        logging.info("Extracting package to temp path")
+        pkg_root_path = self._extract_wheel()
+        pkg_name, version = self._parse_pkg_name()
+        logging.info("package name: {0}, version:{1}".format(pkg_name, version))
+
+        logging.info("Installing package from {}".format(self.pkg_path))
+        self._install_package(pkg_name)
+        logging.info("Generating tokens")
+        apiview = self._generate_tokens(pkg_root_path, pkg_name, version)
+
+        json_out_file_name = "{0}_{1}.json".format(pkg_name, version)
+        out_file_path = os.path.join(pkg_root_path, json_out_file_name)
```

Comment:
This doesn't resolve correctly. When I run it I get 

`C:/repo/scrap/parser_review/temp\azure_storage_blob-12.3.1-py2.py3-none-any\azure-storage
-blob_12.3.1.json`

as the out path. This is because you're not using `self.out_path` appropriately.

Reply:
`out_file_path = os.path.join(self.out_path, json_out_file_name)`